python/pyspark/sql/tests/test_pandas_udf_grouped_map.py

```diff
@@ -29,6 +29,25 @@
     pandas_requirement_message, pyarrow_requirement_message
 from pyspark.testing.utils import QuietTest
 
+if have_pandas:
+    import pandas as pd
+    from pandas.util.testing import assert_frame_equal as pd_assert_frame_equal
+
+if have_pyarrow:
+    import pyarrow as pa
+
+
+def assert_frame_equal(left, right):
+    """
+    Wrap Pandas function because pd.DataFrame.assign will infer mixed types (unicode/str)
+    w/ Python 2, so need to set check_column_type=False
+    """
+    import sys
+    if sys.version < '3':
+        pd_assert_frame_equal(left, right, check_column_type=False)
```

Comment:
This was the only surprise, and it's because calling `DataFrame.apply` with kwargs infers the column names as str for Python 2, and causes DataFrame.columns.inferred_type to be 'mixed' and the assert to fail

Reply:
I don't know, the problem only comes up in these tests because of the way they call `assign`. How about I remove this function wrapping and just make the option conditional for Python version?

Reply:
Ok. Sounds good. Thanks.